### DISCUSSION TITLE
Dynamic mj-button href

### DISCUSSION THREAD

**Discussion by jamesmart77** [mjmlio/mjml#2568] [Q&A] [answered]
Hey all, I cannot figure out how to get dynamic hrefs into my template.

```
<mj-button
      background-color="#78e1e1"
      border-bottom="3px solid #06748c"
      color="#1a1446"
      font-weight="600"
      font-size="16px"
      href="{{${appUrl}}}"
  >
      View App
</mj-button>
```

You can see it now being recognized by the email client...
![image](https://user-images.githubusercontent.com/29242164/199083487-bd8dd86c-6114-4d99-a664-d9e7b8d27130.png)

I can't find anything in the docs...any advice?

**Comment by jamesmart77** [comment, marked as answer]
This was an error on my part. Turns out the solution was as easy as

`href="${appUrl}"` 🤦‍♂️

**Reply by jonathan-messina** [reply]
this didn't work for me, could you give more info about your solution?

**Reply by jamesmart77** [reply]
I am able to do this in tandem with Handlebars.

1. Get the template string and convert to html
![image](https://github.com/mjmlio/mjml/assets/29242164/6d6ebd7b-0d0a-4874-836a-e1279510f44e)

2. Inject the variables into the template
![image](https://github.com/mjmlio/mjml/assets/29242164/554d86ec-2579-40c3-85cc-29f6a71ce4e7)

3. Render template 🚀

**Reply by jonathan-messina** [reply]
Hi @jamesmart77 thanks for the quick response! 
After debugging for a while I found out that my issue was that there was an extra " in my Template string that I feed the variable with.
